libraries/chain/wasm_interface.cpp

```diff
@@ -16,6 +17,7 @@
 namespace eos { namespace chain {
    using namespace IR;
    using namespace Runtime;
+   typedef boost::multiprecision::cpp_dec_float_50 DOUBLE;
```

Comment:
The actual precision of a cpp_dec_float is always slightly higher than the number of digits specified in the template parameter, actually how much higher is an **implementation detail but is always at least 8 decimal** digits.

This indicates that it may not be deterministic. 

The documentation for cpp_bin_float_50 does not have such a disclaimer, that said we should verify that under the hood it only uses integer maths.